build.sbt

```diff
@@ -123,7 +123,7 @@ lazy val observable = project.settings(
 lazy val concurrent =
   project dependsOn core settings (
     defaultSettings,
-    libraryDependencies ++= Seq(catsEffect, catsTagless),
+    libraryDependencies ++= Seq(catsEffect, catsTagless, fs2Core),
```

Comment:
I don't like this dependency, is it necessary?

Reply:
We can implement our own concurrent queue, e.g. Michael-Scott Queue. This would allow us to get rid of fs2

Reply:
@susliko why not scala stdlib's concurrent queue?

Reply:
@atapin Can you be more detailed?

Reply:
@Odomontois I mean scala collections. I mistaken with word `concurrent`